### ISSUE TITLE
Sentinel 2 MSI Level 2A reader

### ISSUE THREAD

**Issue by paalge** [pytroll/satpy#1026]
## Feature Request

Currently there is a reader for Sentinel 2 Level 1C MSI data.
It would be good if this was extended / copied to a new reader supporting the level 2A MSI  data which has been atmosphere corrected.

**Describe the solution you'd like**
Reuse as much as possible from the level 1C reader in order to write a level-2A reader

**Describe any changes to existing user workflow**
Does not affect other parts 

**Additional context**
Currently the alternative is to download the 1C data and process it

**Comment by mraspaud** [comment]
Good idea, thanks for reporting this. This way we could compare the satpy processing the we builtin one :)
